onmt/utils/ExtendedCmdLine.lua

```diff
@@ -209,15 +215,23 @@ function ExtendedCmdLine:dumpConfig(opt, filename)
   local file = assert(io.open(filename, 'w'))
 
   for key, val in pairs(opt) do
-    file:write(key .. ' = ' .. tostring(val) .. '\n')
+    if key:sub(-8) ~= '_default' then
+      file:write(key .. ' = ' .. tostring(val) .. '\n')
+    end
   end
 
   file:close()
 end
 
 function ExtendedCmdLine:parse(arg)
   local i = 1
-  local params = self:default()
+
+  -- set default value
+  local params = {}
+  for option,v in pairs(self.options) do
+    params[strip(option)] = v.default
+    params[strip(option)..'_default'] = true
```

Comment:
It seems simpler and clearer to use a sub-table `params.is_default[strip(option)]` rather than building the key every time.